pyvrp/cpp/bindings.h

```diff
@@ -72,8 +72,16 @@ template <pyvrp::MeasureType T> struct type_caster<pyvrp::Measure<T>>
     bool load(pybind11::handle src,
               [[maybe_unused]] bool convert)  // Python -> C++
     {
-        auto const raw = PyLong_AsLong(src.ptr());
+        if (!convert && !PyLong_Check(src.ptr()))  // strict int when conversion
+            return false;                          // not allowed.
+
+        PyObject *tmp = PyNumber_Long(src.ptr());
+        if (!tmp)
+            return false;
```

Comment:
If `convert` is `false` the Python object needs to be an `int` - no conversion is allowed. When `convert` is `true`, we accept any object `a` for which `int(a)` does not raise.